git-config/tests/file/from_paths/includes/conditional/gitdir/mod.rs

```diff
@@ -0,0 +1,159 @@
+mod util;
+use util::{assert_section_value, git_env_with_symlinked_repo, Condition, GitEnv};
+
+#[test]
+fn relative_path_with_trailing_slash_matches_like_star_star() -> crate::Result {
+    assert_section_value(Condition::new("gitdir:worktree/"), GitEnv::repo_name("worktree")?)
+}
+
+#[test]
+fn relative_path_without_trailing_slash_does_not_match() -> crate::Result {
+    assert_section_value(
+        Condition::new("gitdir:worktree").expect_original_value(),
+        GitEnv::repo_name("worktree")?,
+    )
+}
+
+#[test]
+fn relative_path_without_trailing_slash_and_dot_git_suffix_matches() -> crate::Result {
+    assert_section_value(Condition::new("gitdir:worktree/.git"), GitEnv::repo_name("worktree")?)
+}
+
+#[test]
+fn tilde_slash_expands_the_current_user_home() -> crate::Result {
+    let env = GitEnv::repo_name("subdir/worktree")?;
+    assert_section_value(Condition::new("gitdir:~/subdir/worktree/"), env)
+}
+
+#[test]
+fn tilde_alone_does_not_match_even_if_home_is_git_directory() -> crate::Result {
+    let env = GitEnv::repo_in_home()?;
+    assert_section_value(Condition::new("gitdir:~").expect_original_value(), env)
+}
+
+#[test]
+fn explicit_star_star_prefix_and_suffix_match_zero_or_more_path_components() -> crate::Result {
+    assert_section_value(Condition::new("gitdir:**/worktree/**"), GitEnv::repo_name("worktree")?)
+}
+
+#[test]
+fn dot_slash_path_is_replaced_with_directory_containing_the_including_config_file() -> crate::Result {
+    // TODO: understand this
+    assert_section_value(
+        Condition::new("gitdir:./").set_user_config_instead_of_repo_config(),
+        GitEnv::repo_name("worktree")?,
+    )
+}
+
+#[test]
+#[serial_test::serial]
+#[ignore]
+fn dot_slash_from_environment_causes_error() {
+    use git_config::file::from_paths;
+    // TODO: figure out how to do this, how do we parse sub-keys? Can git do that even?
```

Comment:
@Byron git can do that:

➜  /Users/svetlin/IdeaProjects/gitoxide (cont_include_if_refact) GIT_CONFIG_COUNT=1 GIT_CONFIG_KEY_0=includeIf.gitdir:/Users/svetlin.path GIT_CONFIG_VALUE_0=/Users/svetlin/tmp/a/b/custom-git.cfg ~/CLionProjects/git/giT config --get core.filemode
includeif.gitdir:/Users/svetlin.path
true
➜  /Users/svetlin/IdeaProjects/gitoxide (cont_include_if_refact) GIT_CONFIG_COUNT=1 GIT_CONFIG_KEY_0=includeIf.gitdir:/Users/svetlin/.path GIT_CONFIG_VALUE_0=/Users/svetlin/tmp/a/b/custom-git.cfg ~/CLionProjects/git/giT config --get core.filemode
includeif.gitdir:/Users/svetlin/.path
false

The additional "includeif.gitdir:/Users/svetlin.path" is debug output of what git parsed for the key

Reply:
Thanks for trying it out and letting me know.

To help with scope, I think it's best to leave this as TODO in there, and for good measure I also added another task [in the tracking issue](https://github.com/Byron/gitoxide/issues/331). Eventually this needs to be implemented for correctness, but I don't think it's very pressing right now.